pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -289,13 +319,22 @@ func (r *ReconcileAWSFederatedAccountAccess) createIAMPolicy(awsClient awsclient
 		return &iam.Policy{}, fmt.Errorf("Error marshalling jsonPolicy doc : Error %s", err.Error())
 	}
 
+	var policyName string
+	// Try and build policy name
+	if uuidLabel, ok := afaa.Labels["uuid"]; ok {
+		policyName = afr.Spec.AWSCustomPolicy.Name + "-" + uuidLabel
```

Comment:
Do we need to do any checks here to make sure that that the `afr.Spec.AWSCustomPolicy.Name` is not longer than 128 characters minus the 37 characters from the `-[uuid]`?

Maybe even shorter? (Are we using the Switch Role feature or are we assuming this role through another mechanism?)

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html#w477aac29b9c11b5

Reply:
Yeah, probably should be checked before the code gets to this point, i.e. check at the start of the reconcile and fail if the length is too long.

Reply:
ah, the uuid I'm using is only 6  chars fyi

Reply:
This was a mistake on my part where I thought UUID was a full 36 char UUID string and not just 6 characters.